crates/bevy_asset/src/event.rs

```diff
@@ -0,0 +1,100 @@
+use crate::{Asset, AssetId};
+use bevy_ecs::event::Event;
+use std::fmt::Debug;
+
+/// Events that occur for a specific [`Asset`], such as "value changed" events and "dependency" events.
+#[derive(Event)]
+pub enum AssetEvent<A: Asset> {
+    /// Emitted whenever an [`Asset`] is added.
+    Added { id: AssetId<A> },
+    /// Emitted whenever an [`Asset`] value is modified.
+    Modified { id: AssetId<A> },
+    /// Emitted whenever an [`Asset`] is removed.
+    Removed { id: AssetId<A> },
+    /// Emitted whenever an [`Asset`] has been fully loaded (including its dependencies and all "recursive dependencies").
+    LoadedWithDependencies { id: AssetId<A> },
+}
```

Comment:
It would make this file a bit simpler, but it would make _consuming_ asset events more complicated as consumers would need to do `match asset_event.event_type` instead of `match asset_event` (and know the difference between the types "AssetEvent" and "AssetEventType")  . I prefer the current UX more I think.